### PR TITLE
feat(storage): introduce slice transform mapping

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
for prefix_bloom_filter, introduce slice transform mapping for compute / compact node avoid direct dependencies on table_catalog

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- remove local_table_manager
- use publish_table to replace duplicate notification code
- compute node observer
-  introduce slice transform mapping for compute / compact node

- some fix mention by @yezizp2012 (https://github.com/singularity-data/risingwave/pull/4028)
  - fix compator observer
  - refact notification_service build snapshot

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- https://github.com/singularity-data/risingwave/issues/4027

### CODE REVIEW

File: src/meta/src/manager/notification.rs [hunk 1]
```diff
@@ -242,4 +249,27 @@ impl NotificationManagerCore {
 
         self.current_version
     }
+
+    #[expect(clippy::unused_async)]
+    async fn notify_all(&mut self, operation: Operation, info: &Info) -> NotificationVersion {
```
Comment (yezizp2012) [inline src/meta/src/manager/notification.rs:254]: How about make it non-async directly?

**Review by yezizp2012 — COMMENTED** [review]
The notification part is LGTM.

File: src/meta/src/manager/notification.rs (hunk 1 above)
Comment (Li0k) [reply src/meta/src/manager/notification.rs:254]: yes, i will remove async and the clippy tag

File: risedev.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ risedev:
   # The default configuration will start 1 compute node, 1 meta node and 1 frontend.
   default:
     # If you want to use the local s3 storage, enable the following line
-    # - use: minio
```
Comment (wenym1) [inline risedev.yml:9]: Is changing the default risedev config related to this PR?

File: src/meta/src/manager/catalog.rs [hunk 3]
```diff
@@ -704,6 +656,13 @@ where
             .map(|s| s.id)
             .collect())
     }
+
+    async fn publish_table(&self, operation: Operation, info: Info) -> NotificationVersion {
```
Comment (wenym1) [inline src/meta/src/manager/catalog.rs:660]: Better rename to `broadcast_table_op`

File: src/storage/src/hummock/local_version_manager.rs [hunk 4]
```diff
@@ -213,6 +217,23 @@ impl LocalVersionManager {
         local_version_manager
     }
 
+    pub async fn for_test(
```
Comment (wenym1) [inline src/storage/src/hummock/local_version_manager.rs:220]: We should add a `#[cfg(test, feature = "test")]` to avoid compile it in release mode.

**Review by wenym1 — APPROVED** [review]
Rest LGTM.

**Pull request merged**